server/src/main/java/com/google/musicanalysis/site/AnalysisServlet.java

```diff
@@ -152,4 +137,60 @@ private String getVideoId(String response) {
     // Only returning the first for now, potential to change if I optimize search results.
     return searchResults.get(0);
   }
+
+  /**
+   * Parses the comment JSON string and retrieves the comments.
+   *
+   * @param response The JSON string to be parsed.
+   * @return An ArrayList with each comment
+   */
+  private ArrayList<String> retrieveComments(String response) {
+    ArrayList<String> commentList = new ArrayList<>();
+
+    // Traversing to the items array
+    JsonElement jElement = JsonParser.parseString(response);
+    JsonObject jObject = jElement.getAsJsonObject();
+    JsonArray itemsArray = jObject.getAsJsonArray("items");
+
+    for (JsonElement el : itemsArray) {
+      // Traversing each item in the array
+      String topComment =
+          el.getAsJsonObject()
+              .getAsJsonObject("snippet")
+              .getAsJsonObject("topLevelComment")
+              .getAsJsonObject("snippet")
+              .get("textOriginal")
+              .toString();
+
+      // Remove all non-ASCII characters
+      commentList.add(topComment.replaceAll("[^\\x00-\\x7F]", ""));
+    }
+
+    return commentList;
+  }
+
+  /**
+   * Condenses array of comments into one large string, formatting it along the way.
+   *
+   * @param comments The array to be condensed.
+   * @return A properly formatted Strin
+   */
+  private String convertToString(ArrayList<String> comments) {
+    StringBuilder res = new StringBuilder();
+
+    for (String string : comments) {
+      string = string.replace("\"", "");
+      // Make sure each comment is treated as its own sentence
+      // Not sure char datatype works with regex so used String
+      String lastCharacter = string.substring(string.length() - 1);
+      if (!lastCharacter.matches(".|!|\\?")) {
+        string += ". ";
+      } else {
+        string += " ";
+      }
```

Comment:
so you're just doing the last character because it should be the end of a sentence / word?

Reply:
is comments an array list with each entry being the new line?? I'm just more confused how each line is being split in the array list.

Reply:
Each entry in the array list is a comment from the youtube videos. What i'm doing here is adding each comment to one long string, which I then run through the analyzer. 

Here is my logic

- If the comment does not end in a punctuation mark
    - add a punctuation mark and a space to the large string
-  if the comment ends in a punctuation mark
    - add a space to seperate the comments

I'm doing this in order to make sure that the analyzer reads the string correctly